server/proxy/install.go

```diff
@@ -176,12 +173,34 @@ func (p *Proxy) ensureBot(manifest *apps.Manifest, actingUserID string, client *
 		}
 	}
 
-	token, response := client.CreateUserAccessToken(fullBot.UserId, "Mattermost App Token")
-	if response.StatusCode != http.StatusOK {
-		if response.Error != nil {
-			return nil, nil, response.Error
+	var token *model.UserAccessToken
+	if manifest.RequestedPermissions.Contains(apps.PermissionActAsBot) {
+		user, response = client.GetUser(fullBot.UserId, "")
+		if user == nil {
+			if response.Error != nil {
+				return nil, nil, errors.Wrap(response.Error, "failed to get bot user")
+			}
+			return nil, nil, errors.Errorf("failed to get bot user, status code = %v", response.StatusCode)
+		}
```

Comment:
Nit pick: It's idiomatic to handle errors first 
```suggestion
		if response.Error != nil {
			return nil, nil, errors.Wrap(response.Error, "failed to get bot user")
		}
		if user == nil {
			return nil, nil, errors.Errorf("failed to get bot user, status code = %v", response.StatusCode)
		}
```